sql/core/src/test/resources/sql-tests/inputs/pgSQL/join.sql

```diff
@@ -0,0 +1,2079 @@
+     (select count(1) as cnt, b.a from b group by b.a) as b_grp
+     on a.code = b_grp.a
+  ) as ss
+  on (c.a = ss.code)
+order by c.name;
+
+-- rollback;
+
+--
+-- test incorrect handling of placeholders that only appear in targetlists,
+-- per bug #6154
+--
+SELECT * FROM
+( SELECT 1 as key1 ) sub1
+LEFT JOIN
+( SELECT sub3.key3, sub4.value2, COALESCE(sub4.value2, 66) as value3 FROM
+    ( SELECT 1 as key3 ) sub3
+    LEFT JOIN
+    ( SELECT sub5.key5, COALESCE(sub6.value1, 1) as value2 FROM
+        ( SELECT 1 as key5 ) sub5
+        LEFT JOIN
+        ( SELECT 2 as key6, 42 as value1 ) sub6
+        ON sub5.key5 = sub6.key6
+    ) sub4
+    ON sub4.key5 = sub3.key3
+) sub2
+ON sub1.key1 = sub2.key3;
+
+-- test the path using join aliases, too
+SELECT * FROM
+( SELECT 1 as key1 ) sub1
+LEFT JOIN
+( SELECT sub3.key3, value2, COALESCE(value2, 66) as value3 FROM
+    ( SELECT 1 as key3 ) sub3
+    LEFT JOIN
+    ( SELECT sub5.key5, COALESCE(sub6.value1, 1) as value2 FROM
+        ( SELECT 1 as key5 ) sub5
+        LEFT JOIN
+        ( SELECT 2 as key6, 42 as value1 ) sub6
+        ON sub5.key5 = sub6.key6
+    ) sub4
+    ON sub4.key5 = sub3.key3
+) sub2
+ON sub1.key1 = sub2.key3;
+
+--
+-- test case where a PlaceHolderVar is used as a nestloop parameter
+--
+
+-- EXPLAIN (COSTS OFF)
+-- SELECT qq, unique1
+--   FROM
+--   ( SELECT COALESCE(q1, 0) AS qq FROM int8_tbl a ) AS ss1
+--   FULL OUTER JOIN
+--   ( SELECT COALESCE(q2, -1) AS qq FROM int8_tbl b ) AS ss2
+--   USING (qq)
+--   INNER JOIN tenk1 c ON qq = unique2;
+
+SELECT qq, unique1
+  FROM
+  ( SELECT COALESCE(q1, 0) AS qq FROM int8_tbl a ) AS ss1
+  FULL OUTER JOIN
+  ( SELECT COALESCE(q2, -1) AS qq FROM int8_tbl b ) AS ss2
+  USING (qq)
+  INNER JOIN tenk1 c ON qq = unique2;
+
+--
+-- nested nestloops can require nested PlaceHolderVars
+--
+
+create or replace temporary view nt1 as select * from
+  (values(1,true,true), (2,true,false), (3,false,false))
+  as v(id, a1, a2);
+create or replace temporary view nt2 as select * from
+  (values(1,1,true,true), (2,2,true,false), (3,3,false,false))
+  as v(id, nt1_id, b1, b2);
+create or replace temporary view nt3 as select * from
+  (values(1,1,true), (2,2,false), (3,3,true))
+  as v(id, nt2_id, c1);
+-- explain (costs off)
+-- select nt3.id
+-- from nt3 as nt3
+--   left join
+--     (select nt2.*, (nt2.b1 and ss1.a3) AS b3
+--      from nt2 as nt2
+--        left join
+--          (select nt1.*, (nt1.id is not null) as a3 from nt1) as ss1
+--          on ss1.id = nt2.nt1_id
+--     ) as ss2
+--     on ss2.id = nt3.nt2_id
+-- where nt3.id = 1 and ss2.b3;
+
+select nt3.id
+from nt3 as nt3
+  left join
+    (select nt2.*, (nt2.b1 and ss1.a3) AS b3
+     from nt2 as nt2
+       left join
+         (select nt1.*, (nt1.id is not null) as a3 from nt1) as ss1
+         on ss1.id = nt2.nt1_id
+    ) as ss2
+    on ss2.id = nt3.nt2_id
+where nt3.id = 1 and ss2.b3;
+
+-- [SPARK-28379] Correlated scalar subqueries must be aggregated
```

Comment:
nit. The JIRA title seems to be just an error message.

Reply:
Yes. I do not know the root cause.